hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/AutoRecordKeyGenerationUtils.scala

```diff
@@ -54,7 +51,9 @@ object AutoRecordKeyGenerationUtils {
     }
   }
 
-  def isAutoGenerateRecordKeys(parameters: Map[String, String]): Boolean = {
-    !parameters.contains(KeyGeneratorOptions.RECORDKEY_FIELD_NAME.key()) // if record key is not configured,
+  def shouldAutoGenerateRecordKeys(parameters: Map[String, String]): Boolean = {
+    val recordKeyFromTableConfig = parameters.getOrElse(HoodieTableConfig.RECORDKEY_FIELDS.key(), "")
+    val recordKeyFromWriterConfig = parameters.getOrElse(KeyGeneratorOptions.RECORDKEY_FIELD_NAME.key(), "")
+    recordKeyFromTableConfig.isEmpty && recordKeyFromWriterConfig.isEmpty
```

Comment:
There is no need to check two times, the `parameters` is inferred and merged from hoodie table config already.

Reply:
this gives more flexibility without assuming `parameters`, right?